internal/generate/clusterserviceversion/bases/definitions/crd_test.go

```diff
@@ -0,0 +1,208 @@
+		}
+		out = getTypedDescriptors(markedFields, reflect.TypeOf(v1alpha1.SpecDescriptor{}), spec)
+		Expect(out).To(HaveLen(0))
+	})
+	It("returns one status descriptor for one status marker on a field", func() {
+		markedFields[""] = []*fieldInfo{
+			{
+				FieldInfo: markers.FieldInfo{
+					Markers: markers.MarkerValues{
+						"": []interface{}{
+							Descriptor{"status", "Foo", []string{"urn:alm:descriptor:com.tectonic.ui:text"}, nil},
+						},
+					},
+				},
+			},
+		}
+		out = getTypedDescriptors(markedFields, reflect.TypeOf(v1alpha1.StatusDescriptor{}), status)
+		Expect(out).To(HaveLen(1))
+		Expect(out).To(BeEquivalentTo([]interface{}{
+			v1alpha1.StatusDescriptor{
+				DisplayName:  "Foo",
+				XDescriptors: []string{"urn:alm:descriptor:com.tectonic.ui:text"},
+			},
+		}))
+	})
+	It("returns one spec descriptor for three spec markers and one status marker on a field", func() {
+		markedFields[""] = []*fieldInfo{
+			{
+				FieldInfo: markers.FieldInfo{
+					Markers: markers.MarkerValues{
+						"": []interface{}{
+							Descriptor{"spec", "Foo", nil, nil},
+							Descriptor{"spec", "", nil, intPtr(2)},
+							Descriptor{"spec", "", []string{"urn:alm:descriptor:com.tectonic.ui:text"}, nil},
+							Descriptor{"status", "", []string{"urn:alm:descriptor:com.tectonic.ui:arrayFieldGroup:blah"}, nil},
+						},
+					},
+				},
+				pathSegments: []string{"foo", inlinedTag, "bar", "baz"},
+			},
+		}
+		out = getTypedDescriptors(markedFields, reflect.TypeOf(v1alpha1.SpecDescriptor{}), spec)
+		Expect(out).To(HaveLen(1))
+		Expect(out).To(BeEquivalentTo([]interface{}{
+			v1alpha1.SpecDescriptor{
+				DisplayName:  "Foo",
+				XDescriptors: []string{"urn:alm:descriptor:com.tectonic.ui:text"},
+				Path:         "foo.bar.baz",
+			},
+		}))
+	})
+	It("returns two spec descriptor for spec markers on two different fields", func() {
+		markedFields[""] = []*fieldInfo{
+			{
+				FieldInfo: markers.FieldInfo{
+					Markers: markers.MarkerValues{
+						"": []interface{}{Descriptor{"spec", "Foo", nil, intPtr(1)}},
+					},
+				},
+				pathSegments: []string{"foo"},
+			},
+			{
+				FieldInfo: markers.FieldInfo{
+					Markers: markers.MarkerValues{
+						"": []interface{}{Descriptor{"spec", "Bar", nil, intPtr(0)}},
+					},
+				},
+				pathSegments: []string{"bar"},
+			},
+		}
+		out = getTypedDescriptors(markedFields, reflect.TypeOf(v1alpha1.SpecDescriptor{}), spec)
+		Expect(out).To(HaveLen(2))
+		Expect(out).To(BeEquivalentTo([]interface{}{
+			v1alpha1.SpecDescriptor{DisplayName: "Bar", Path: "bar"},
+			v1alpha1.SpecDescriptor{DisplayName: "Foo", Path: "foo"},
+		}))
+	})
+	It("returns multiple sorted spec descriptors with all orders set", func() {
+		markedFields, expected := makeMockMarkedFields()
+		out = getTypedDescriptors(markedFields, reflect.TypeOf(v1alpha1.SpecDescriptor{}), spec)
+		Expect(out).To(HaveLen(len(expected)))
+		Expect(out).To(BeEquivalentTo(expected))
+	})
+})
+
+func intPtr(i int) *int { return &i }
```

Comment:
This is probably my biggest pet peeve about go having to create these util function for primitive pointers.

Reply:
Agreed.